src/invariant_service.rs

```diff
@@ -893,39 +891,21 @@ where
         InvariantStorage::as_ref().tickmap_slice(min_chunk_index..=max_chunk_index, pool_key)
     }
 
-    pub fn get_liquidity_ticks(&self, pool_key: PoolKey) -> Vec<LiquidityTick> {
-        let mut ticks = vec![];
-        let tick_spacing = pool_key.fee_tier.tick_spacing;
-
-        let max_tick = get_max_tick(tick_spacing);
-        let (chunk_limit, bit_limit) = tick_to_position(max_tick, tick_spacing);
-
-        let invariant = InvariantStorage::as_ref();
-        for i in 0..=chunk_limit {
-            let chunk = invariant.tickmap.bitmap.get(&(i, pool_key)).unwrap_or(&0);
-
-            if chunk != &0 {
-                let end = if *chunk as u16 == chunk_limit {
-                    bit_limit
-                } else {
-                    (CHUNK_SIZE - 1) as u8
-                };
-
-                for bit in 0..=end {
-                    if get_bit_at_position(*chunk, bit) == 1 {
-                        let tick_index = position_to_tick(i, bit, tick_spacing);
-
-                        invariant
-                            .ticks
-                            .get(pool_key, tick_index)
-                            .map(|tick| ticks.push(LiquidityTick::from(tick)))
-                            .unwrap();
-                    }
-                }
-            }
+    pub fn get_liquidity_ticks(
+        &self,
+        pool_key: PoolKey,
+        tickmap: Vec<i32>,
```

Comment:
What about the limit of `Vec`? Did you check if there are any limits when passing arguments to the entrypoint?

Reply:
Yes, the limit is 3MB the same for response and request size, with  current LIQUIDITY_TICK_LIMIT this Vec fits within that limit